go/vt/vtgate/planbuilder/testdata/union_cases.json

```diff
@@ -1490,40 +1490,62 @@
         "TableName": "`user`_music",
         "Inputs": [
           {
-            "OperatorType": "Distinct",
-            "Collations": [
-              "(0:1)"
+            "OperatorType": "SimpleProjection",
+            "ColumnNames": [
+              "foo"
+            ],
+            "Columns": [
+              0
             ],
             "Inputs": [
```

Comment:
the original query does not have `foo` as alias, I think this can be avoided here.